src/context/state-context.js

```diff
@@ -0,0 +1,25 @@
+import { createContext, useContext, useReducer } from "react";
+
+import { stateReducer } from "./state-reducer";
+
+const CartContext = createContext();
+
+export const CartContextProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(stateReducer, {
+    products: [],
+    itemsInCart: [],
+    itemsInWishlist: [],
+    route: "home",
+    sortBy: "",
+    includeOutOfStock: true,
+    filterByCategories: [],
+    filterByBrands: []
```

Comment:
Can create a common object for these as in future filters can inc a lot so club them as a single object

Reply:
Yes definitely, i ll update this.